### PR TITLE
disabled widget based on lists

### LINKED ISSUE #269: Disable widgets when they cant be used in a given context
campus form - step 4 - qualification titles
image (3)

There are two lists: 
- on the left; qualifications already added to the campus
- on the right; all the qualifications

If the left list is empty, the widget `remove` must be disabled. 
If the right list is empty, the widget `add` must be disabled.

If both lists are not empty, then both widgets are active.

### PR DESCRIPTION
**Pull request by boney-bun**
fix #269

### CODE REVIEW

**Comment by boney-bun** [comment]
PIA:
ford3#278

**Comment by pierrealixt** [comment]
@boney-bun 
By disabled, I meant changing the color of the arrow and the text. Maybe in grey.

File: django_project/ford3/static/js/campus_form_qualifications.js [hunk 1]
```diff
@@ -56,6 +56,37 @@ const setClickEventToLi = () => {
...
 }
 
+//isEnabled true = disabled add button
+const toggleAddButton = (isEnabled) => {
+  addButton = getAddQualificationButtonElement()
+  addButton.disabled = isEnabled
+}
+
+const toggleRemoveButton = (isEnabled) => {
+  removeButton = getRemoveQualificationButtonElement()
```
Comment (pierrealixt) [inline django_project/ford3/static/js/campus_form_qualifications.js]: Please install https://standardjs.com/

it should `const removeButton ...`

File: django_project/ford3/static/js/campus_form_qualifications.js [hunk 2]
```diff
@@ -56,6 +56,37 @@ const setClickEventToLi = () => {
   })
 }
 
+//isEnabled true = disabled add button
+const toggleAddButton = (isEnabled) => {
+  addButton = getAddQualificationButtonElement()
```
Comment (pierrealixt) [inline django_project/ford3/static/js/campus_form_qualifications.js]: `const addButton ...`

File: django_project/ford3/static/js/campus_form_qualifications.js [hunk 3]
```diff
@@ -56,6 +56,37 @@ const setClickEventToLi = () => {
...
+}
+
+const toggleRemoveButton = (isEnabled) => {
+  removeButton = getRemoveQualificationButtonElement()
+  removeButton.disabled = isEnabled
+}
+
+const checkSaqaList = () =>
+{
+    saqaList = getSaqaQualificationsListElement()
```
Comment (pierrealixt) [inline django_project/ford3/static/js/campus_form_qualifications.js]: `const saqaList ...`

File: django_project/ford3/static/js/campus_form_qualifications.js [hunk 4]
```diff
@@ -56,6 +56,37 @@ const setClickEventToLi = () => {
...
+    if (saqaList.length == 0 || saqaList.getElementsByTagName("li").length < 1) {
+        toggleAddButton(true);
+    }
+}
+
+const checkCampusList = () => {
+  campusList = getCampusQualificationsListElement()
```
Comment (pierrealixt) [inline django_project/ford3/static/js/campus_form_qualifications.js]: `const campusList ...`

**Comment by boney-bun** [comment]
@pierrealixt what do you think of this?

ford3#278_rev

**Comment by timlinux** [comment]
@boney-bun looking better. That box around the arrow on click still looks ugly. Could you change it so that it shows a bolder thicker version of the arrow on click and then goes back to the skinny version after the interaction is done please?

**Comment by boney-bun** [comment]
@tim i made those bold arrow buttons myself.
let me know if you think it's not good enough.
below is with the bold arrows effect:
ford3#278_rev

**Review by Christiaanvdm — APPROVED** [review]
LGTM

**Pull request merged**